test/Microsoft.ML.Tests/TrainerEstimators/MatrixFactorizationTests.cs

```diff
@@ -331,5 +331,99 @@ public void MatrixFactorizationInMemoryDataZeroBaseIndex()
+                nameof(OneClassMatrixElementZeroBased.MatrixRowIndex),
+                nameof(OneClassMatrixElementZeroBased.Value),
+                advancedSettings: s =>
+                {
+                    s.Fun = MatrixFactorizationTrainer.LibMFLossFunctionType.SquareLossOneClass;
+                    s.NumIterations = 100;
+                    s.NumThreads = 1; // To eliminate randomness, # of threads must be 1.
+                    // Let's test non-default regularization coefficient.
+                    s.Lambda = 0.025;
+                    s.K = 16;
+                    // Importance coefficient of loss function over matrix elements not specified in the input matrix.
+                    s.Alpha = 0.01;
+                    // Desired value for matrix elements not specified in the input matrix.
+                    s.C = 0.15;
+                });
+
+            // Train a matrix factorization model.
+            var model = pipeline.Fit(dataView);
+
+            // Apply the trained model to the training set.
+            var prediction = model.Transform(dataView);
+
+            // Calculate regression matrices for the prediction result.
+            var metrics = mlContext.Regression.Evaluate(prediction, label: "Value", score: "Score");
+
+            // Make sure the prediction error is not too large.
+            Assert.InRange(metrics.L2, 0, 0.0016);
+
+            // Create data for testing. Note that the 2nd element is not specified in the training data so it should
+            // be close to the constant specified by s.C = 0.15. Comparing with the data structure used in training phase,
+            // one extra float is added into OneClassMatrixElementZeroBasedForScore for storing the prediction result. Note
+            // that the prediction engine may ignore the Value and assign the predicted value to Score.
+            var testDataMatrix = new List<OneClassMatrixElementZeroBasedForScore>();
+            testDataMatrix.Add(new OneClassMatrixElementZeroBasedForScore() { MatrixColumnIndex = 0, MatrixRowIndex = 0, Value = 0, Score = 0 });
+            testDataMatrix.Add(new OneClassMatrixElementZeroBasedForScore() { MatrixColumnIndex = 1, MatrixRowIndex = 2, Value = 0, Score = 0 });
+
+            // Convert the in-memory matrix into an IDataView so that ML.NET components can consume it.
+            var testDataView = ComponentCreation.CreateDataView(Env, testDataMatrix);
+
+            // Apply the trained model to the test data.
+            var testPrediction = model.Transform(testDataView);
+
+            double tolerance = Math.Pow(10, -6);
+            var testResults = new List<OneClassMatrixElementZeroBasedForScore>(testPrediction.AsEnumerable<OneClassMatrixElementZeroBasedForScore>(mlContext, false));
+            // Positive example (i.e., examples can be found in dataMatrix) is close to 1.
+            Assert.InRange(testResults[0].Score, 0.982391 - tolerance, 0.982391 + tolerance);
```

Comment:
>Assert.InRange [](start = 12, length = 14)

you care use CompareNumbersWithTolerance

Reply:
Sure.